addons/osfstorage/models.py

```diff
@@ -365,6 +367,7 @@ def on_add(self):
         if self.root_node:
```

Comment:
@sloria, here's the problem. Modm would have nulled out `root_node` upon clone. So, getting here, `self.root_node` would have been falsely. So it would have made a new one based off the `self.owner` of the `NodeSettings`, which would have correctly been set to the `Registration` in `after_register()`, seen below. 

I was thinking about comparing the types of `self.owner` and `self.root_node` and then nulling out `root_node` if the types mismatched, but I don't know if that's sufficient.

@chrisseto, other than nulling out foreign fields on create, which @sloria was hoping to avoid, can you think of a better way to do this?